src/core/level_selection_system.cpp

```diff
@@ -0,0 +1,100 @@
+#include "platform/i_platform.h"
+#include "platform/filesystem.h"
+#include "level_selection_system.h"
+#include "core/level_selected_event.h"
+
+namespace core {
+
+LevelSelectionSystem::LevelSelectionSystem()
+    : mLevelModel( "level", &RootModel::Get() )
+    , mSelectLevelModel( IntFunc( this, &LevelSelectionSystem::SelectLevel ), "select", &mLevelModel )
+    , mGetSelectedLevelModel( GetStringFunc( this, &LevelSelectionSystem::GetSelectedLevel), "getselected", &mLevelModel ) 
```

Comment:
A sample that can be used:
, mLevelModel( (ModelValue::get_string_vec_t) RefTo(mLevelNames),"levels", &mEditorModel )

this way you don't need a getter func.